.github/workflows/ci.yml

```diff
@@ -30,29 +30,36 @@ jobs:
                     slepc: 'slepc'
 
         steps:
-        -   uses: actions/checkout@v3
-        -   name: Set up Python ${{ matrix.python }}
-            uses: actions/setup-python@v4
+        -   uses: actions/checkout@v4
+        -   name: Set up Conda
+            uses: conda-incubator/setup-miniconda@v3.1.0
             with:
                 python-version: ${{ matrix.python }}
+                auto-update-conda: true
+                channels: conda-forge,defaults
 
         -   name: Install pip dependencies
+            shell: bash -el {0}
             run: |
                 python -m pip install --upgrade pip
-                pip install tox codecov
+                pip install '.[test]'
+
+        -   name: Install conda dependencies
+            if: ${{ matrix.slepc == 'slepc' }}
+            shell: bash -el {0}
+            run: |
+                conda install petsc slepc petsc4py slepc4py mpi4py
 
         -   name: Test
+            shell: bash -el {0}
             run: |
-                tox -e py${{ matrix.python }}-${{ matrix.slepc }} -vv
-            env:
-                PLATFORM: ${{ matrix.os }}
+                python -m pytest --cov --cov-config=tox.ini --no-cov-on-fail --cov-report=xml --cov-report=term-missing:skip-covered -vv
 
         -   name: Upload coverage
-            uses: codecov/codecov-action@v3
+            uses: codecov/codecov-action@v4
             with:
+                fail_ci_if_error: false
                 files: ./coverage.xml
-                flags: unittests
                 name: ${{ matrix.os }}-${{ matrix.python }}-${{ matrix.slepc }}
-                env_vars: OS,PYTHON
-                fail_ci_if_error: false
+                token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
@msmdev could you please create and add this token to the repo secrets?

Reply:
Seems that it was already there in the repo secrets...